lib/src/json_schema/validator.dart

```diff
@@ -782,6 +802,18 @@ class Validator {
     return lastFound;
   }
 
+  /// A helper function to deal with infinite loops at evaluation time.
+  /// If we see the same data/ref pair twice, we're in a loop.
+  void _withRefScope(Uri refScope, Instance instance, RefScopeOperation fn) {
```

Comment:
Will this business slow things down for earlier drafts?

Reply:
Added a ticket to the backlog for perf testing: https://jira.atl.workiva.net/browse/SCM-778